Sources/SpelledPitch/PitchSpeller/Wetherfield/Wetherfield.swift

```diff
@@ -90,30 +90,26 @@ struct PitchSpeller {
         }
 
         return assignedNodes
-            .sorted {
-                switch (
-                    $0.index.a < $1.index.a,
-                    $0.index.a == $1.index.a,
-                    UInt8($0.index.b.rawValue) < UInt8($1.index.b.rawValue)
-                ) {
-                case (true, _, _): return true
-                case (_, true, true): return true
-                default: return false
+            .reduce(into: [Int: (AssignedNode, AssignedNode)]()) { pairs, node in
+                if !pairs.keys.contains(node.index.a) {
+                    pairs[node.index.a] = (node, node)
                 }
-            }
-            .pairs
-            .map(spellPitch)
+                switch node.index.b {
+                case .up: pairs[node.index.a]!.0 = node
+                case .down: pairs[node.index.a]!.1 = node
+                }
+            }.mapValues(spellPitch)
     }
 
     private func spellPitch(_ up: AssignedNode, _ down: AssignedNode) -> SpelledPitch<EDO12> {
-        let pitch = self.pitch(node: up.index)
+        let pitch = self.pitch(node: up.index)!
         let tendencies = TendencyPair((up.assignment, down.assignment))
         let spelling = Pitch.Spelling<EDO12>(pitchClass: pitch.class, tendencies: tendencies)!
         return SpelledPitch(try! pitch.spelled(with: Pitch.Spelling<EDO48>(spelling)))
     }
 
     /// - Returns: The `Pitch` value for the given `node` value.
-    private func pitch(node: Cross<Int, Tendency>) -> Pitch {
+    private func pitch(node: Cross<Int, Tendency>) -> Pitch? {
```

Comment:
If this is `private`, maybe we could just return a `Pitch` (non-`Optional`), as that would perhaps be a programmer error, and something that should never even be possible. Thus you could:

```Swift
return pitches[node.a]!
```